src/rust/rust-basic.ts

```diff
@@ -7,41 +7,40 @@ import {
 } from "./visitors/interface_visitor.js";
 import { EnumVisitor } from "./visitors/enum_visitor.js";
 import { UnionVisitor } from "./visitors/union_visitor.js";
-import { rustifyCaps } from "./utils/index.js";
-import { apexToRustType } from "./utils/types.js";
-import { visibility } from "./utils/config.js";
+import { rustifyCaps, types } from "./utils/index.js";
+import { visibility } from "./utils/index.js";
 
 export class RustBasic extends ContextWriter {
   constructor(writer: Writer) {
     super(writer);
   }
 
   visitType(context: Context): void {
-    this.append(new StructVisitor(context).toString());
+    this.append(new StructVisitor(context.type, context).toString());
```

Comment:
Why not only pass in `context` here? Obviously `type` is enclosed in `context`.

Reply:
The waPC codegen had a spot where it dynamically created an AST node from a callback and instantiated a visitor from there. Rather than decipher the intent was, it was easier and more extensible to make the AST node arg explicit.